app/controllers/badges/badges.js

```diff
@@ -0,0 +1,195 @@
+const moment = require('moment');
+const cfg = require('config');
+const shortid = require('shortid');
+const api = require('../../lib/api');
+const mongo = require('../../lib/mongo');
+const cache = require('../../lib/cache');
+const logger = require('../../lib/logger');
+
+exports.badges = (req, res) => {
+  const { id } = req.query;
+
+  mongo.get('wallet').findOne({ userId: id }, (err, result) => {
+    if (err) {
+      return res.json({ status: 'error', data: 'NO_BADGES' });
+    }
+
+    const { badges } = cache.get();
+
+    if (!result || !result.badges) {
+      return res.json({ status: 'success', data: { badges } });
+    }
+
+    const userBadges = [];
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const badge of badges) {
```

Comment:
A `map()` would be more idiomatic than pushing in a loop, and wouldn't require to disable `no-restricted-syntax`